src/autopas/cells/SortedCellView.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file FullSortedParticleCell.h
+ * @date 18.01.2018
+ * @author C. Menges
+ */
+
+#pragma once
+
+#include <vector>
+#include "autopas/cells/FullParticleCell.h"
+#include "autopas/cells/ParticleCell.h"
+#include "autopas/iterators/SingleCellIterator.h"
+#include "autopas/utils/ArrayMath.h"
+#include "autopas/utils/CudaSoA.h"
+#include "autopas/utils/SoA.h"
+#include "autopas/utils/WrapOpenMP.h"
+
+namespace autopas {
+
+/**
+ * This class defines a sorted view on a given ParticleCell. Particles are sorted along the normalized vector r. The
+ * projected position of a particle as well as a pointer to the underlying particle are stored in a vector, which is
+ * sorted by the projected positions.
+ * @note Insertion, deletion and change of position of particles invalidates the view.
+ * @tparam Particle
+ */
+template <class Particle, class ParticleCellType>
+class SortedCellView : public ParticleCell<Particle> {
+ public:
+  /**
+   * Constructs a FullSortedParticleCell.
+   * @param cell Cell whose particles are sorted.
+   * @param r Normalized vector along particles are sorted.
+   */
+  SortedCellView(ParticleCellType &cell, const std::array<double, 3> &r) : _cell(&cell) {
+    _particles.reserve(cell.numParticles());
+    for (auto p = getStaticCellIter(cell); p.isValid(); ++p) {
+      _particles.push_back(std::make_pair(ArrayMath::dot(p->getR(), r), &(*p)));
+    }
+    std::sort(_particles.begin(), _particles.end(),
+              [](const auto &a, const auto &b) -> bool { return a.first < b.first; });
+  }
+
+  void addParticle(Particle &m) override {}
+
+  virtual SingleCellIteratorWrapper<Particle> begin() override { return _cell->begin(); }
```

Comment:
```suggestion
  SingleCellIteratorWrapper<Particle> begin() override { return _cell->begin(); }
```
I think you could remove this virtual?